codes/c/chapter_backtracking/preorder_traversal_ii_compact.c

```diff
@@ -6,62 +6,50 @@
 
 #include "../utils/common.h"
 
+// 假设结果长度不超过 100
+#define MAX_SIZE 100
+
+TreeNode *path[MAX_SIZE];
+TreeNode *res[MAX_SIZE][MAX_SIZE];
+int pathSize = 0, resSize = 0;
+
```

Comment:
Define an array to represent the length of each path.
But it's not a big problem when you see the traversal results later on.
`for (int j = 0; res[i][j] != NULL; ++j) {`